op-program/client/l1/blob_fetcher.go

```diff
@@ -0,0 +1,37 @@
+package l1
+
+import (
+	"context"
+	"errors"
+
+	"github.com/ethereum-optimism/optimism/op-node/rollup/derive"
+	"github.com/ethereum-optimism/optimism/op-service/eth"
+	"github.com/ethereum/go-ethereum/log"
+)
+
+var InvalidHashesLengthError = errors.New("invalid hashes length")
+
+type BlobFetcher struct {
+	logger log.Logger
+	oracle Oracle
+}
+
+var _ = (*derive.L1BlobsFetcher)(nil)
+
+func NewBlobFetcher(logger log.Logger, oracle Oracle) *BlobFetcher {
+	return &BlobFetcher{
+		logger: logger,
+		oracle: oracle,
+	}
+}
+
+// GetBlobs fetches blobs that were confirmed in the given L1 block with the given indexed hashes. It does so by fetching
+// the 4096 field elements for each blob through the preimage oracle.
+func (b *BlobFetcher) GetBlobs(ctx context.Context, ref eth.L1BlockRef, hashes []eth.IndexedBlobHash) ([]*eth.Blob, error) {
+	blobs := make([]*eth.Blob, len(hashes))
+	for i := 0; i < len(hashes); i++ {
+		b.logger.Info("Fetching blob", "l1_ref", ref.Hash, "blob_versioned_hash", hashes[i].Hash, "index", hashes[i].Index)
```

Comment:
Do we expect the number of fetched blobs to be significantly large? If so this may become too verbose and we should log that we are fetching `len(hashes)` blobs instead.

Reply:
Nope; Number of blobs should be small. The blob limit on L1 is pretty low, believe target = 3 & limit = 6.